src/classes/HiFiAudioAPIData.ts

```diff
@@ -32,6 +32,188 @@ export class Point3D {
         this.y = y;
         this.z = z;
     }
+
+    /**
+     * Returns the vector dot product between two Point3Ds.
+     */
+    static dot(a: Point3D, b: Point3D): number {
+        return a.x * b.x + a.y * b.y + a.z * b.z;
+    }
+
+    /**
+     * Returns the vector cross product between two Point3Ds.
+     */
+    static cross(a: Point3D, b: Point3D): Point3D {
+        return new Point3D({x: a.y*b.z - a.z*b.y, y: a.z*b.x - a.x*b.z, z: a.x*b.y - a.y*b.x});
+    }
+
+    /**
+      * Negates this Point3D.
+      */
+    negate() {
+        this.x = -this.x;
+        this.y = -this.y;
+        this.z = -this.z;
+    }
+}
```

Comment:
The **Point3D** `dot`, `cross`, and `negate` methods above were necessary to do the math below.